rector-php.php

```diff
@@ -119,6 +130,36 @@
                 'getName',
                 new UnionType([new NullType(), new StringType()]),
             ),
+            new AddReturnTypeDeclaration(
+                ListingsItem::class,
+                'getAttributes',
+                new UnionType([new NullType(), new ArrayType(new MixedType(), new MixedType())]),
+            ),
+        ]
+    );
+
+    /**
+     * Contrary to Amazon's schema, some of the properties are in fact nullable - this rule will fix that.
+     */
+    $config->ruleWithConfiguration(
+        SetNullableFunctionReturnTypeRector::class,
```

Comment:
why you can't use AddReturnTypeDeclarationRector? It seems to work in some places, no?

Reply:
related to https://github.com/rectorphp/rector/issues/7395 - changing type to the less precise one is not supported.